File: solutions_11-20/Euler_18.py
# Find the maximum total from top to bottom of the triangle below:

import os
import timeit


start = timeit.default_timer()

def euler_18():
    with open(os.path.abspath('').strip('solutions_11-20') + 'euler_txt/triangle1.txt') as f:
        x = list()
        for line in f.readlines():
            x.append(line.split())
    tot = 10
    row = 0
    position = 0
    tot += int(x[row][position])
    row += 1
    while row < len(x):
        if int(x[row][position]) > int(x[row][position+1]):
            tot += int(x[row][position])
        else:
            tot += int(x[row][position+1])
            position += 1
        row += 1
    return tot

print "Answer: %s" % euler_18()
stop = timeit.default_timer()
print "Time: %f" % (stop - start)

File: Perl/perl_solutions_11-20/Euler_18.pl
use strict;
use warnings;
use Time::HiRes qw( clock );


sub tri_path
{
    my @triangle;
    my $path = "/home/tijko/Project_Euler/euler_txt/triangle1.txt";
    open (my $tri, $path);
    while (my $i = <$tri>) {
        chomp($i);
        my @row = split(" ", $i);
        @triangle = (@triangle, \@row);
    }
    close $tri;
    my $total = $triangle[0][0] + 10;
    my $col = 0;
    my $row = 1;
    while ($row < scalar(@triangle)) {
        if ($triangle[$row][$col] < $triangle[$row][$col + 1]) {
            $total += $triangle[$row][$col + 1];
            $col += 1;
            $row += 1;
            next;
        }
        $total += $triangle[$row][$col];
        $row += 1;
    }
    return $total;
}

my $start = clock();
my $answer = tri_path();
my $stop = clock();
print "Answer: $answer\n";
printf "Time: %.5f\n", $stop - $start;
